src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -56,14 +56,23 @@ private BoundDeconstructionAssignmentOperator BindDeconstructionAssignment(CShar
 
             if ((object)boundRHS.Type == null)
```

Comment:
When RHS is in error, its type is `ErrorType` (not null) which is already handled below (by `DeconstructIntoStep`). Added a test explicitly for this.